### PR TITLE
[feature fix] Update tests to use URLs relative to API_PREFIX

### LINKED ISSUE #3062: Update all tests to use URLs relative to API_BASE
All API unit tests currently point to URLs of the form `v2/<endpoint>`, which is a form of hard-coding and could break if the URL prefix in Django ever changes.

These URLs should be modified to add the prefix based on `osf.io/api/base/settings/defaults.py: API_BASE`. 

This would affect anything that lives under `tests/api_tests`.

From brianjgeiger#109

### PR DESCRIPTION
**Pull request by jamiehand**
Closes #3062 

### Purpose

All API unit tests currently point to URLs of the form `v2/<endpoint>`, which is a form of hard-coding and could break if the URL prefix in Django ever changes. This PR changes the prefix from hard-coded to being relative to `osf.io/api/base/settings/defaults.py: API_PREFIX` (note that API_PREFIX is based on API_BASE, which is referenced in the issue this fixes, #3062).

### Changes

Change prefix of URLs in `tests/api_tests` from `/v2/` to `API_PREFIX`.

### CODE REVIEW

**Comment by abought** [comment]
An alternate solution would be to use `website.util.api_v2_url`. This would take care of some edge cases, and ensure that the correct settings file variables were used to construct the URL. (even if we changed what the variable was called)

To make this work with unit tests, your new url would be as follows:

`url = api_v2_url(url_fragment, base_route='')`

**Comment by abought** [comment]
Also, if you do decide to go with hard-coded URLs, the variable `API_BASE` might be preferable. Basically, `API_BASE` is the part of the URL that Django (and the test runner) know about; `API_PATH` is the part that gets put in front of Django by an external server program. The variable `API_PREFIX` combines these two.

Running locally, `API_PATH` is blank, but that's not the case in production. (where you access the api at `api/v2`) It may be best to not treat the variables as the same thing.

...Honestly, these variables are subtle, complicated, and sometimes confusingly named. Sorry about that. :)

**Comment by jamiehand** [comment]
Thanks @abought! Actually @brianjgeiger ended up suggesting that I not use api_v2_url (because of potential conflicts with another feature being changed right now, if I understood correctly) and also said that I should use API_PREFIX instead of API_BASE (I originally was using API_BASE). Brian, do you think I should change this based on Andy's comments? (e.g. I didn't realize about the edge cases.)

**Comment by brianjgeiger** [comment]
The reason we didn't go with the helper function is that the unit tests don't use the domain and port as part of the request, it's all relative from root. I'm not opposed to giving it a try and using it if it works, but I don't know if it will cause problems with the tests or not. But API_PREFIX should be correct under all circumstances.

**Comment by brianjgeiger** [comment]
Okay, @abought chatted with me, and I've given terrible advice. Use API_BASE instead, as he suggested.

File: tests/api_tests/nodes/test_views.py [hunk 1]
```diff
@@ -16,7 +17,7 @@ def setUp(self):
         self.user.set_password('justapoorboy')
         self.user.save()
         self.basic_auth = (self.user.username, 'justapoorboy')
-        self.url = '/v2/'
+        self.url = '/' + API_BASE
```
Comment (brianjgeiger) [inline tests/api_tests/nodes/test_views.py]: This will be a general change, but for strings, we prefer to manipulate them using format(). https://cosdev.readthedocs.org/en/latest/style_guides/python.html#string-formatting. Other than that it looks good.

**Comment by brianjgeiger** [comment]
Pass done. :penguin:

File: tests/api_tests/nodes/test_views.py (hunk 1 above)
Comment (jamiehand) [inline tests/api_tests/nodes/test_views.py]: Oh okay. Thanks!

**Comment by brianjgeiger** [comment]
This looks great. :tada: :penguin: :tada: @sloria (or @lyndsysimon since he's assigned himself) can doublecheck.

**Comment by lyndsysimon** [comment]
Looks great, ready for @sloria.

I did a comprehensive search over the `/tests/api_tests/` directory, and verified that nothing was missed. I also made sure the changes pass the more restricting flake8 settings that have been merge ahead of this (they do!).

:gem:

**Comment by brianjgeiger** [comment]
So unfortunately we found a flaw in understanding, so expect a new PR about this in the near future. Nothing is broken now, but it's not fixing the right thing, so deployment is not going to be what it's supposed to be.
